tuning/sft_trainer.py

```diff
@@ -103,25 +92,9 @@ def train(
         None for fine tuning
             The peft configuration to pass to trainer
     """
-    run_distributed = int(os.environ.get("WORLD_SIZE", "1")) > 1
 
     logger = logging.get_logger("sft_trainer")
 
-    # Validate parameters
-    if (not isinstance(train_args.num_train_epochs, float)) or (
-        train_args.num_train_epochs <= 0
-    ):
-        raise ValueError("num_train_epochs has to be an integer/float >= 1")
-    if (not isinstance(train_args.gradient_accumulation_steps, int)) or (
-        train_args.gradient_accumulation_steps <= 0
-    ):
-        raise ValueError("gradient_accumulation_steps has to be an integer >= 1")
-
-    # make sure to unset FSDP args when running on single gpu
-    if not run_distributed:
-        train_args.fsdp = ""
```

Comment:
does single GPU still work after this change? without `train_args.fsdp_config = {"xla": False}` I remember single GPU training was failing

Reply:
Yes I have verified that i runs in both single and multi gpu settings. 

if I can point your attention to [`TrainingArguments.__post_init__`](https://github.com/huggingface/transformers/blob/45c065109074d60c587d3e562f16531d02a422f6/src/transformers/training_args.py#L1691-L1707), you will see that it already has logic to assemble all these `self.fsdp_config` flags appropriately. 

As [it has been coming up](https://github.com/foundation-model-stack/fms-hf-tuning/pull/79#discussion_r1513617678) that we should keep `transformers < 4.38.0`, the reason is because at `4.38` they introduced a new `xla_fsdp_v2 ` flag, and it did not suffice to only set the `xla` flag. Hence, we should leave this to the `__post_init__`  

My guess is why it did not run before was that before this changes, the `__post_init__` of `TrainingArguments` is not called, and perhaps the flag `fsdp_config["xla"]` is

Reply:
thank you!

Reply:
actually post_init is always called as it is a derived class. I think the error I was referring to was because in original first commit we were oevrriding it with the wrong value . we were doing `
        train_args.fsdp_config = {}` or something like that, so I changed it to `"xla": False` . But you are right that if we dont do anything the _post_init should take care of it. So makes sense to remove it, thanks